### ISSUE TITLE
Reproducing NDCG@100 score on MovieLens 20M

### ISSUE THREAD

**Issue by Knorreman** [recombee/ELSA#1]
Hi,

I have implemented ELSA and I seem to get around 0.55 NDCG@100 instead of 0.426 as you have in the paper. So I am wondering if there is something else I am missing? It seems wrong. I used the ndcg_score function from scikit: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.ndcg_score.html

I use as you wrote in the paper:
Batch size: 32
Latent factors: 800
NAdam with 0.1 learning rate
10 epochs
Convert rating of 4 or higher as an interaction

Thanks,

**Comment by zombak79** [comment]
Hi!

Thanks for trying ELSA. I will prepare scripts and notebooks to reproduce results on MovieLens20M dataset in separate branch. The problem is that initially, we implemented ELSA in tensorflow and used a framework for validation from this paper  https://arxiv.org/pdf/2102.05774.pdf implemented in this repo https://github.com/zombak79/vasp. But tensorflow implementation had several production issues, so we later implemented ELSA in pytorch. Implementation presented here is (except small details) the same model that runs in Recombee production. 

Reproducibility branch will be ready in few hours :)

Vojtech.

**Comment by zombak79** [comment]
Code for reproducing the results on the MovieLens20M dataset is in the branch  `reproduce_movielens` https://github.com/recombee/ELSA/tree/reproduce_movielens.

I will look into the issue with NDCG from sklearn. We used implementation from https://github.com/dawenl/vae_cf, and we have no experiences with the sklearn implementation yet.

**Comment by Knorreman** [comment]
Thanks, I will have a look at the branch and the NDCG implementation! :)
